clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -200,28 +200,25 @@ bool CheckedRegionFinder::VisitCallExpr(CallExpr *C) {
 }
 
 bool CheckedRegionFinder::VisitVarDecl(VarDecl *VD) {
-  auto CVSet = Info.getVariable(VD, Context);
-  Wild = isWild(CVSet) || containsUncheckedPtr(VD->getType());
+  CVarOption CV = Info.getVariable(VD, Context);
+  Wild = isWild(CV) || containsUncheckedPtr(VD->getType());
   return true;
 }
 
 bool CheckedRegionFinder::VisitParmVarDecl(ParmVarDecl *PVD) {
   // Check if the variable is WILD.
-  auto CVSet = Info.getVariable(PVD, Context);
-  Wild |= isWild(CVSet) | containsUncheckedPtr(PVD->getType());
+  CVarOption CV = Info.getVariable(PVD, Context);
+  Wild |= isWild(CV) || containsUncheckedPtr(PVD->getType());
   return true;
 }
 
 bool CheckedRegionFinder::VisitMemberExpr(MemberExpr *E){
   ValueDecl *VD = E->getMemberDecl();
   if (VD) {
     // Check if the variable is WILD.
-    std::set<ConstraintVariable *> CVSet = Info.getVariable(VD, Context);
-    for (auto Cv : CVSet) {
-      if (Cv->hasWild(Info.getConstraints().getVariables())) {
-        Wild = true;
-      }
-    }
+    CVarOption Cv = Info.getVariable(VD, Context);
+    if (Cv && (*Cv)->hasWild(Info.getConstraints().getVariables()))
```

Comment:
This is another place I'm confused about `Option<ConstraintVariable *>` -- is the check to `Cv` whether the option is empty or not? What if the contents of the option are `nullptr` ?

Reply:
An unfortunate problem I ran into is that I can't write `Option<ConstraintVariable>` because `ConstraintVariable` is an abstract class. I also can't store `ConstraintVariable &` for some other reason I don't fully understand. The only other option is to store `ConstraintVariable *`, but this leaves the possibility that contents are `nullptr`. This isn't great, but I still think it's better than passing around a raw pointer because the `Option` type at least acts as an explicit annotation that the constraint variable might be missing.

Reply:
@Machiry  You might be able to comment on this if you're more familiar with C++ than we are. What I would like to do is store the abstract class `ConstraintVariable` in the `llvm::Optional`` type, but this is not permitted by C++. Is there a workaround that doesn't require storing a pointer?